### PR TITLE
Add release notes for 1.75.0

### PR DESCRIPTION
**Pull request by cuviper**
cc @rust-lang/release
r? @Mark-Simulacrum

### CODE REVIEW

File: RELEASES.md [hunk 1]
```diff
@@ -1,3 +1,128 @@
...
+
+<a id="1.75.0-Rustdoc"></a>
+
+Rustdoc
+-------
+
+- [Accept less invalid Rust in rustdoc.](https://github.com/rust-lang/rust/pull/117450/)
+- [Document lack of object safety on affected traits.](https://github.com/rust-lang/rust/pull/113241/)
+- [Hide `#[repr(transparent)]` if it isn't part of the public ABI.](https://github.com/rust-lang/rust/pull/115439/)
+- [Show enum discrimant if it is a C-like variant.](https://github.com/rust-lang/rust/pull/116142/)
```
Comment (fmease) [inline RELEASES.md:100]:
```suggestion
- [Show enum discriminant if it is a C-like variant.](https://github.com/rust-lang/rust/pull/116142/)
```

typo carried over from PR title

File: RELEASES.md [hunk 2]
```diff
@@ -1,3 +1,128 @@
...
+Internal Changes
+----------------
+
+These changes do not affect any public interfaces of Rust, but they represent
+significant improvements to the performance or internals of rustc and related
+tools.
+
+- [Optimize `librustc_driver.so` with BOLT.](https://github.com/rust-lang/rust/pull/116352/)
+- [Enable parallel rustc front end in dev and nightly builds.](https://github.com/rust-lang/rust/pull/117435/)
+- [Distribute `rustc-codegen-cratelift` as rustup component on the nightly channel.](https://github.com/rust-lang/rust/pull/81746/)
```
Comment (Mark-Simulacrum) [inline RELEASES.md:124]:
```suggestion
- [Distribute `rustc-codegen-cranelift` as rustup component on the nightly channel.](https://github.com/rust-lang/rust/pull/81746/)
```

Comment (cuviper) [reply RELEASES.md:124]: Whoops!

File: RELEASES.md [hunk 3]
```diff
@@ -1,3 +1,128 @@
+Version 1.75.0 (2023-12-28)
+==========================
+
+<a id="1.75.0-Language"></a>
+
+Language
+--------
+
+- [Stabilize `async fn` and return-position `impl Trait` in traits.](https://github.com/rust-lang/rust/pull/115822/)
+- [Allow function pointer signatures containing `&mut T` in `const` contexts.](https://github.com/rust-lang/rust/pull/116015/)
```
Comment (ChrisDenton) [inline RELEASES.md:10]: I dunno if this is fixable but when skimming the notes I saw "`&mut T` in `const` contexts" and got momentarily confused/excited until I re-read the start of the line.

File: RELEASES.md [hunk 4]
```diff
@@ -1,3 +1,128 @@
...
+Libraries
+---------
+
+- [Override `Waker::clone_from` to avoid cloning `Waker`s unnecessarily.](https://github.com/rust-lang/rust/pull/96979/)
+- [Implement `BufRead` for `VecDeque<u8>`.](https://github.com/rust-lang/rust/pull/110604/)
+- [Implement `FusedIterator` for `DecodeUtf16` when the inner iterator does.](https://github.com/rust-lang/rust/pull/110729/)
+- [Implement `Not, Bit{And,Or}{,Assign}` for IP addresses.](https://github.com/rust-lang/rust/pull/113747/)
+- [Implement `Default` for `ExitCode`.](https://github.com/rust-lang/rust/pull/114589/)
+- [Guarantee representation of None in NPO](https://github.com/rust-lang/rust/pull/115333/)
+- [Don't panic in `<BorrowedCursor as io::Write>::write`](https://github.com/rust-lang/rust/pull/115460/)
```
Comment (a1phyr) [inline RELEASES.md:46]: This is a nightly API so I think that it could be removed

**Review by a1phyr — COMMENTED** [review]
Also shouldn't #116505 be noted somewhere ? (Probably Compiler or Internals) This feels quite an notable change to me.

**Comment by pietroalbini** [comment]
Applied both of a1phyr's points.

@bors r+ rollup

**Pull request merged**